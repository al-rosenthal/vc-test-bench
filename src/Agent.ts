export type ConnectionRef = {connection_id: string}
export type Invitation = {invitation_url: string} & ConnectionRef


export interface Agent {
    createInvitationToConnect(): Promise<Invitation>
    receiveInvitation(invitation: Invitation): ConnectionRef;
}
