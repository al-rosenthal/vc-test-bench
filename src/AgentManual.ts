import { AcceptProofArgs, AriesAgent, ConnectionRef, CredentialOfferRef, Invitation, ReceiveInvitationResponse } from "./Agent";
import { Agent, ConnectionRecord, ConnectionsModule, CredentialExchangeRecord, CredentialsModule, Logger, MediationRecipientModule, ProofsModule, V2CredentialProtocol, V2ProofProtocol } from "@credo-ts/core";
import { AnonCredsCredentialFormatService, AnonCredsModule, AnonCredsProofFormatService, LegacyIndyCredentialFormatService, LegacyIndyProofFormatService, V1CredentialProtocol, V1ProofProtocol } from "@credo-ts/anoncreds";
import { AskarModule } from "@credo-ts/askar";
import { IndyVdrModule } from "@credo-ts/indy-vdr";
import { CredentialDefinitionBuilder, ProofRequestBuilder, SchemaBuilder } from "./lib";
import QRCode from 'qrcode'
import readline from 'readline';
import fs from 'node:fs';
import path from 'node:path';

export class AgentManual implements AriesAgent {
    config: any;
    logger: Logger;
    public constructor(config:any,logger: Logger){
        this.config = config
        this.logger = logger
    }
  sendOOBConnectionlessProofRequest(builder: ProofRequestBuilder): Promise<any | undefined> {
    throw new Error("Method not implemented.");
  }
  acceptProof(proof: AcceptProofArgs): Promise<void> {
    //throw new Error("Method not implemented.");
    return Promise.resolve()
  }
  waitForPresentation(presentation_exchange_id: string): Promise<void> {
    throw new Error("Method not implemented.");
  }
  sendConnectionlessProofRequest(builder: ProofRequestBuilder): Promise<string | undefined> {
    throw new Error("Method not implemented.");
  }
    async acceptCredentialOffer(offer: CredentialOfferRef): Promise<void> {
      console.warn('Accept Credential')
    }
    async findCredentialOffer(connectionId: string): Promise<CredentialOfferRef> {
      return {id: 'undefined', connection_id: connectionId}
    }
    createSchemaCredDefinition(credDefBuilder: CredentialDefinitionBuilder): Promise<string | undefined> {
      throw new Error("Method not implemented.");
    }
    createSchema(builder: SchemaBuilder): Promise<string | undefined> {
        throw new Error("Method not implemented.");
    }
    createInvitationToConnect(): Promise<Invitation> {
        throw new Error("Method not implemented.");
    }
    async receiveInvitation(invitationRef: Invitation): Promise<ReceiveInvitationResponse> {
      const relativePath = '/Library/Android/sdk/emulator/resources/custom.png'
      const QRCodePath = path.join(process.env.HOME as string, relativePath)
      fs.mkdirSync(path.dirname(QRCodePath), { recursive: true })
      fs.writeFileSync(path.join(process.cwd(), 'invitation_url.txt'), invitationRef.invitation_url)
      fs.writeFileSync(path.join(process.cwd(), 'invitation.json'), JSON.stringify((invitationRef as any)))
      await QRCode.toFile(
            QRCodePath,
            invitationRef.invitation_url,
            {margin: 10}
      )
      return {}
    }
    public async startup(){
    }
}