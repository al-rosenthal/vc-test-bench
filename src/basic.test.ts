import { describe, test } from "@jest/globals";
import { AgentTraction } from "./AgentTraction";
import { AgentCredo } from "./AgentCredo";
import { LogLevel } from "@credo-ts/core";
import { PersonCredential1, PersonSchemaV1_1 } from "./mocks";
import { CredentialDefinitionBuilder, issueCredential, PinoLogger, ProofRequestBuilder, RequestAttributeBuilder, seconds_since_epoch, verifyCredentialA1, verifyCredentialA2, verifyCredentialB1, verifyCredentialB2, waitFor, withRedirectUrl } from "./lib";
import pino from "pino";

const stepTimeout = 120_000
const shortTimeout = (2*60)*1000

import { setGlobalDispatcher, Agent} from 'undici';
import { AriesAgent, ResponseCreateInvitationV1 } from "./Agent";
import { AgentManual } from "./AgentManual";
setGlobalDispatcher(new Agent({connect: { timeout: 20_000 }}));

export const loggerTransport = pino.transport({
  targets: [
    {
      level: 'trace',
      target: 'pino/file',
      options: {
        destination: `./logs/run-${Date.now()}.log.ndjson`,
        autoEnd: true,
      },
    }
  ],
})

describe("Mandatory", () => {
  const _logger = pino({ level: 'trace', timestamp: pino.stdTimeFunctions.isoTime, }, loggerTransport);
  const logger = new PinoLogger(_logger, LogLevel.trace)
  const config = require("../local.env.json")["sovrin_testnet"];
  const ledgers = require("../ledgers.json");
  const agentA = new AgentTraction(config, logger);
  //const agentB: AriesAgent = new AgentManual(config, new ConsoleLogger(LogLevel.trace))
  const agentB: AriesAgent =  process.env.HOLDER_TYPE === 'manual'? new AgentManual(config, logger): new AgentCredo(config,ledgers, logger)
  //new PinoLogger(logger, LogLevel.trace));
  const schema = new PersonSchemaV1_1();
  const credDef = new CredentialDefinitionBuilder()
    .setSchema(schema)
    .setSupportRevocation(true);
  
  beforeAll(async () => {
    logger.info('1 - beforeAll')
    await agentA.startup()
    await agentB.startup()
    //await Promise.all([agentA.startup(), agentB.startup()]);

  }, stepTimeout);
  afterAll(async () => {
    logger.info('1 - afterAll')
    await agentB.shutdown();
    _logger.flush();
    //loggerTransport.end();
  }, stepTimeout);
  test("connected/v1/M1", async () => {
    const issuer = agentA
    const holder = agentB
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const remoteInvitation = await issuer.createInvitationToConnect() as ResponseCreateInvitationV1
    logger.info(`waiting for holder to accept connection`)
    const agentBConnectionRef1 = await holder.receiveInvitation(remoteInvitation)
    logger.info(`waiting for issuer to accept connection`)
    await issuer.waitForConnectionReady(remoteInvitation.payload.connection_id as string)
    logger.info(`${remoteInvitation.payload.connection_id} connected to ${agentBConnectionRef1.connectionRecord?.connection_id}`)
    logger.info('agentBConnectionRef1', agentBConnectionRef1)
    const msgSent: any = await issuer.sendBasicMessage(remoteInvitation.payload.connection_id as string, 'Hello')
    logger.info('Message Sent:', msgSent)
    await waitFor(10_000)
    await holder.sendBasicMessage(agentBConnectionRef1.connectionRecord?.connection_id as string, 'ok')
    const msgRcvd = await issuer.waitForBasicMessage(remoteInvitation.payload.connection_id as string, Date.parse(msgSent.created_at as string), ["k", "ok"])
    logger.info('Message Received:', msgRcvd)
  }, shortTimeout);
  test.skip("OOB/connected/messaging", async () => {
    const issuer = agentA
    const holder = agentB
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const remoteInvitation = await issuer.createOOBInvitationToConnect()
    logger.info(`waiting for holder to accept connection`)
    const agentBConnectionRef1 = await holder.receiveInvitation(remoteInvitation)
    logger.info(`waiting for issuer to accept connection`)
    const {connection_id} =  await issuer.waitForOOBConnectionReady(remoteInvitation.payload.invi_msg_id)
    logger.info(`${connection_id} connected to ${agentBConnectionRef1.connectionRecord?.connection_id}`)
    logger.info('agentBConnectionRef1', agentBConnectionRef1)
    const msgSent: any = await issuer.sendBasicMessage(connection_id, 'Hello')
    logger.info('Message Sent:', msgSent)
    await waitFor(10_000)
    await holder.sendBasicMessage(agentBConnectionRef1.connectionRecord?.connection_id as string, 'ok')
    const msgRcvd = await issuer.waitForBasicMessage(connection_id, Date.parse(msgSent.created_at as string), ["k", "ok"])
    logger.info('Message Received:', msgRcvd)
  }, shortTimeout);
  test("setup", async () => {
    logger.info(`Executing ${expect.getState().currentTestName}`)

    try{
      await agentA.createSchema(schema);
      await agentA.createSchemaCredDefinition(credDef);
      await agentA.clearAllRecords()
      
      const personCred = new PersonCredential1(credDef)

      await issueCredential(agentA, agentB, personCred)
    }catch (error){
      console.dir(error)
      throw error
    }
  }, stepTimeout)
  test("connectionless/present-proof-1.0/encoded-payload", async () => {
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )
    await verifyCredentialA1(agentA, agentB, proofRequest)
  }, shortTimeout);
  test("connectionless/present-proof-1.0/url-redirect", async () => {
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )
    await verifyCredentialA2(agentA, agentB, proofRequest)
  }, shortTimeout);
  test("connectionless/present-proof-2.0/encoded-payload", async () => {
    const verifier = agentA
    const holder = agentB
    const { logger } = verifier
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )
    const remoteInvitation2 = await verifier.sendConnectionlessProofRequestV2(proofRequest)
    const agentBConnectionRef2 = await holder.receiveInvitation(remoteInvitation2)
    //console.dir(['agentBConnectionRef', agentBConnectionRef2])
    if (agentBConnectionRef2.invitationRequestsThreadIds){
      for (const proofId of agentBConnectionRef2.invitationRequestsThreadIds) {
        await holder.acceptProof({id: proofId})
      }
    }
    logger.info('remoteInvitation2', remoteInvitation2)
    await verifier.waitForPresentationV2(remoteInvitation2.payload.presentation_exchange_id as string)
  }, shortTimeout);
  test("connectionless/present-proof-2.0/url-redirect", async () => {
    const verifier = agentA
    const holder = agentB
    const { logger } = verifier
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )
    const remoteInvitation2 = await withRedirectUrl(await verifier.sendConnectionlessProofRequestV2(proofRequest))
    const agentBConnectionRef2 = await holder.receiveInvitation(remoteInvitation2)
    //console.dir(['agentBConnectionRef', agentBConnectionRef2])
    if (agentBConnectionRef2.invitationRequestsThreadIds){
      for (const proofId of agentBConnectionRef2.invitationRequestsThreadIds) {
        await holder.acceptProof({id: proofId})
      }
    }
    await verifier.waitForPresentationV2(remoteInvitation2.payload.presentation_exchange_id as string)
  }, shortTimeout);
  test("OOB/connectionless/present-proof-1.0/encoded-payload", async () => {
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )
    await verifyCredentialB1(agentA, agentB, proofRequest)
  }, shortTimeout);
  test("OOB/connectionless/present-proof-1.0/url-redirect", async () => {
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )
    await verifyCredentialB2(agentA, agentB, proofRequest)
  }, shortTimeout);
  test("OOB/connectionless/present-proof-2.0/encoded-payload", async () => {
    const verifier = agentA
    const holder = agentB
    const { logger } = verifier
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )

    const remoteInvitation3 = await verifier.sendOOBConnectionlessProofRequestV2(proofRequest)
    logger.info('remoteInvitation3', remoteInvitation3)
    logger.info(`Holder is receiving invitation for ${remoteInvitation3.payload.presentation_exchange_id}`)
    const agentBConnectionRef3 = await holder.receiveInvitation(remoteInvitation3)
    logger.info('Holder is accepting proofs')
    //await waitFor(10000)
    if (agentBConnectionRef3.invitationRequestsThreadIds){
      for (const proofId of agentBConnectionRef3.invitationRequestsThreadIds) {
        await holder.acceptProof({id: proofId})
      }
    }
    logger.info(`Verifier is waiting for proofs: ${remoteInvitation3.payload.presentation_exchange_id}`)
    await verifier.waitForPresentationV2(remoteInvitation3.payload.presentation_exchange_id  as string)
  }, shortTimeout);
  test("OOB/connectionless/present-proof-2.0/url-redirect", async () => {
    const verifier = agentA
    const holder = agentB
    const { logger } = verifier
    logger.info(`Executing ${expect.getState().currentTestName}`)
    const proofRequest = new ProofRequestBuilder()
        .addRequestedAttribute("studentInfo",
            new RequestAttributeBuilder()
                .setNames(["given_names", "family_name"])
                //.addRestriction({"cred_def_id": credDef.getId()})
                .addRestriction({"schema_name": schema.getName(),"schema_version": schema.getVersion(),"issuer_did": credDef.getId()?.split(':')[0]})
                .setNonRevoked(seconds_since_epoch(new Date()))
    )

    const remoteInvitation3 = await withRedirectUrl(await verifier.sendOOBConnectionlessProofRequestV2(proofRequest))
    logger.info('remoteInvitation3', remoteInvitation3)
    logger.info(`Holder is receiving invitation for ${remoteInvitation3.payload.presentation_exchange_id}`)
    const agentBConnectionRef3 = await holder.receiveInvitation(remoteInvitation3)
    logger.info('Holder is accepting proofs')
    //await waitFor(10000)
    if (agentBConnectionRef3.invitationRequestsThreadIds){
      for (const proofId of agentBConnectionRef3.invitationRequestsThreadIds) {
        await holder.acceptProof({id: proofId})
      }
    }
    logger.info(`Verifier is waiting for proofs: ${remoteInvitation3.payload.presentation_exchange_id}`)
    await verifier.waitForPresentationV2(remoteInvitation3.payload.presentation_exchange_id as string)
  }, shortTimeout);
});
