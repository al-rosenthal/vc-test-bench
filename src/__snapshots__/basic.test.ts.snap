// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mandatory connected/present-proof-1.0/encoded-payload 1`] = `
[
  {
    "baseUrl": "--redacted--",
    "data": {
      "image_url": "https://bc-wallet-demo-agent-admin.apps.silver.devops.gov.bc.ca/public/student/connection/best-bc-logo.png",
      "my_label": "Faber\`s ðŸ˜‡ - {timestamp}",
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "auto_remove": false,
      "auto_verify": true,
      "comment": "string",
      "connection_id": "{connection_id}",
      "cred_def_id": "{cred_def_id}",
      "proof_request": "--redacted--",
      "trace": false,
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "params": {},
    "url": "/present-proof/send-request",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/present-proof/records/{record_id}",
  },
]
`;

exports[`Mandatory connected/v1/M1 1`] = `
[
  {
    "baseUrl": "--redacted--",
    "data": {
      "image_url": "https://bc-wallet-demo-agent-admin.apps.silver.devops.gov.bc.ca/public/student/connection/best-bc-logo.png",
      "my_label": "Faber\`s ðŸ˜‡ - {timestamp}",
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "content": "Hello",
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "url": "/connections/{connection_id}/send-message",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {
      "connection_id": "{connection_id}",
      "state": "sent",
    },
    "url": "/basicmessages",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {
      "connection_id": "{connection_id}",
      "state": "received",
    },
    "url": "/basicmessages",
  },
]
`;

exports[`Mandatory connectionless/present-proof-1.0/encoded-payload 1`] = `
[
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/wallet/did/public",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "auto_remove": false,
      "auto_verify": true,
      "comment": "string",
      "proof_request": "--redacted--",
      "trace": false,
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "params": {},
    "url": "/present-proof/create-request",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/present-proof/records/{record_id}",
  },
]
`;

exports[`Mandatory connectionless/present-proof-1.0/url-redirect 1`] = `
[
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/wallet/did/public",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "auto_remove": false,
      "auto_verify": true,
      "comment": "string",
      "proof_request": "--redacted--",
      "trace": false,
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "params": {},
    "url": "/present-proof/create-request",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/present-proof/records/{record_id}",
  },
]
`;

exports[`Mandatory connectionless/present-proof-2.0/encoded-payload 1`] = `
[
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/wallet/did/public",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "auto_remove": false,
      "auto_verify": true,
      "comment": "string",
      "presentation_request": "--redacted--",
      "trace": false,
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "params": {},
    "url": "/present-proof-2.0/create-request",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/present-proof-2.0/records/{record_id}",
  },
]
`;

exports[`Mandatory setup 1`] = `
[
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {
      "schema_name": "Person",
      "schema_version": "1.1",
    },
    "url": "/schemas/created",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {
      "schema_name": "Person",
      "schema_version": "1.1",
    },
    "url": "/credential-definitions/created",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/credential-definitions/{cred_def_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/present-proof/records",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "delete",
    "params": {},
    "url": "/present-proof/records/{record_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/present-proof/records",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/issue-credential/records",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "delete",
    "params": {},
    "url": "/issue-credential/records/{record_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/issue-credential/records",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/connections",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "delete",
    "params": {},
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "params": {},
    "url": "/connections",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "image_url": "https://bc-wallet-demo-agent-admin.apps.silver.devops.gov.bc.ca/public/student/connection/best-bc-logo.png",
      "my_label": "Faber\`s ðŸ˜‡ - {timestamp}",
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/connections/{connection_id}",
  },
  {
    "baseUrl": "--redacted--",
    "data": {
      "auto_issue": true,
      "auto_remove": false,
      "connection_id": "{connection_id}",
      "cred_def_id": "{cred_def_id}",
      "credential_preview": "--redacted--",
      "trace": true,
    },
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "post",
    "url": "/issue-credential/send-offer",
  },
  {
    "baseUrl": "--redacted--",
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Authorization": "--redacted--",
      "Content-Type": "application/json",
    },
    "method": "get",
    "url": "/issue-credential/records/{record_id}",
  },
]
`;
